stubs/docutils/docutils/statemachine.pyi

```diff
@@ -1,3 +1,194 @@
+    nested_sm_kwargs: dict[str, Any] | None
+    transition_order: list[str]
+    transitions: dict[str, tuple[Pattern[str], Callable[[], None], str]]
+    state_machine: StateMachine
+    debug: bool
+    def __init__(self, state_machine: StateMachine, debug: bool = False) -> None: ...
+    def runtime_init(self) -> None: ...
+    def unlink(self) -> None: ...
+    def add_initial_transitions(self) -> None: ...
+    def add_transitions(self, names: list[str], transitions) -> None: ...
+    def add_transition(self, name: str, transition: tuple[Pattern[str], str, str]) -> None: ...
+    def remove_transition(self, name: str) -> None: ...
+    def make_transition(self, name: str, next_state: str | None = None) -> tuple[Pattern[str], _TransitionMethod, str]: ...
+    def make_transitions(
+        self, name_list: list[str | tuple[str] | tuple[str, str]]
+    ) -> tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod, str]]]: ...
+    def no_match(
+        self, context: list[str], transitions: tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod, str]]]
+    ) -> _TransitionResult: ...
+    def bof(self, context: list[str]) -> tuple[list[str], list[str]]: ...
+    def eof(self, context: list[str]) -> list[str]: ...
+    def nop(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+
+class StateMachineWS(StateMachine):
+    def get_indented(self, until_blank: bool = False, strip_indent: bool = True) -> tuple[StringList, int, int, bool]: ...
+    def get_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True
+    ) -> tuple[list[str], int, bool]: ...
+    def get_first_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True, strip_top: bool = True
+    ) -> tuple[list[str], int, int, bool]: ...
+
+class StateWS(State):
+    indent_sm: type[StateMachine] | None
+    indent_sm_kwargs: dict[str, Any] | None
+    known_indent_sm: type[StateMachine] | None
+    known_indent_sm_kwargs: dict[str, Any] | None
+    ws_patterns: dict[str, str]
+    ws_initial_transitions: Sequence[str]
+    def __init__(self, state_machine: StateMachine, debug: bool = False) -> None: ...
+    def add_initial_transitions(self) -> None: ...
+    def blank(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+    def indent(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+    def known_indent(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+    def first_known_indent(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+
+class _SearchOverride:
+    def match(self, pattern: Pattern[str]) -> Match[str]: ...
+
+class SearchStateMachine(_SearchOverride, StateMachine): ...
+class SearchStateMachineWS(_SearchOverride, StateMachineWS): ...
+
+_Data = TypeVar("_Data")
```

Comment:
Nit: I'd use `_T` as name, similar to most collection types in typeshed are typed. An move the type var to the top of the file, seeing as how generic it is.